cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/SvInjectionConversion.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.cgmes.conversion.elements;
+
+import com.powsybl.cgmes.conversion.Context;
+import com.powsybl.cgmes.model.CgmesTerminal;
+import com.powsybl.iidm.network.*;
+import com.powsybl.triplestore.api.PropertyBag;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public class SvInjectionConversion extends AbstractIdentifiedObjectConversion {
+
+    public SvInjectionConversion(PropertyBag p, Context context) {
+        super("SvInjection", p, context);
+        String tn = p.getId("TopologicalNode");
+        Terminal associatedTerminal = context.terminalMapping().findFromTopologicalNode(tn);
+        if (associatedTerminal == null) {
```

Comment:
The logic to find/create terminal related data and iidm node is not clear, maybe we should extract the different options to separate functions with proper names?

Reply:
I refactored the code to make it clearer, is it enough?